lib/ImageWorker.js

```diff
@@ -34,33 +29,72 @@ const wrappedComponent = WrappedComponent => props => {
   return <WrappedComponent {...props} />;
 };
 
+/** Have we initiated the worker pool already? */
+const workerPoolCreated = false;
+const workerPool = [];
+
+function createWorkerPool() {
+  const blobURL = URL.createObjectURL(
+    new Blob([webWorkerScript], { type: 'application/javascript' })
+  );
+  for (let i = 0; i < window.navigator.hardwareConcurrency || 4; ++i) {
```

Comment:
This will go into an infinite loop

Reply:
Why so?

Reply:
when you say 
```javascript
i < window.navigator.hardwareConcurrency || 4
```

The browser starts evaluating from the left and this will be treated as
`(i < window.navigator.hardwareConcurrency) || (4)`
even if the value of `i` goes to say 200, the above will then become 
`200 < window.navigator.hardwareConcurrency || 4`
Which will translate to `false || 4` and will always be true